droidlet/dialog/chat_schema.md

```diff
@@ -0,0 +1,71 @@
+## Description ##
+This is the source of truth for the most updated version of the schema to be followed when sending chats from the agent the the user (and perhaps eventually from the agent to other agents).  When updating this document please increment the version number.  Older versions of the schema can be found in git history.
+
+Note that chats sent from the user to the agent should always be in plaintext, as all messages should be parsed by the NSP and interpreted.  I.e. special pathways and conditionals on agent input are discouraged.  It is possible to help the NSP by including special tokens that are reserved for specific use cases, eg. clarification response.
+
+### Content Types ###
+ - `chat_string`: Only text to display in the chat window with no response assumed.
+    - Example: "I finished building this!"
+ - `chat_and_media`: Text and media to display in the chat window, with no response assumed.
+    - Example: "I finished building this!" (with accompanying photo)
+ - `chat_and_text_options`: Text to display in the chat window along with a list of possible response options.
+    - Example: "Are these the Droidlets you are looking for?" ('no', 'no')
+ - `chat_and_media_options`: Text to display in the chat window along with a list of clickable media options.
+    - Example: "Which of these is the 'blue sphere' you were referring to?" (pictures of blue spheres)
+ - `chat_and_media_and_text_options`: Text and media to display in the chat window alond with a list of possible response options.
+    - Example: "Is this funny?" <img src='https://external-preview.redd.it/hWh_8TpqrT6zAwpzHJ_m9Rx3iHjc_yI4zSI6aazMFTc.jpg?auto=webp&s=6d8006ac3edca5bad98dd7b5b9a4a8d5554eaff0' width="200"> ('yes', 'no')
+
+### Schema ###
+
+<pre>
+{
+	"title": "Droidlet Chat",
+	"description": "A single chat sent from a Droidlet agent to the user",
+	"version": 1,
+	"type": "object",
+
+	"properties": {
+		"speaker_id": {
+			"description": "The memid of the speaker",
+			"type": "string"
+		},
+
+		"timestamp": {
+			"description": "Unix timestamp of when the chat was sent",
+			"type": "integer"
+		},
+
+		"content_type": {
+			"description": "The type of content to be rendered to the user",
+			"type": "array",
+			"items": {
+				"enum": [
+					"chat_string",
+					"chat_and_media",
+					"chat_and_text_options",
+					"chat_and_media_options",
+					"chat_and_media_and_text_options"
+				]
+			},
+			"minItems": 1,
+			"maxItems": 1
+		},
+
+		"content": {
```

Comment:
We should not limit the response options to 2. There are cases where the user may have to select from a larger number of options.

Reply:
Since the inner array is specified exactly the min/max 2 is actually irrelevant, I think, so I deleted it for clarity.  However just so we're in the same page, the format of this field is an array of arrays where the outer array has length >= 1 and the inner arrays consist of two items, essentially a key:value pair.  Here's an example of what the content field could look like:

`[ ["text", "Is this the cube you want me to destroy?"], ["response_option", "yes"], ["response_option", "no"] ]`

Reply:
Thanks for the clarification. In that case, why not use a JSON Object (ie KVP) instead of an array?